sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -239,6 +239,21 @@ class SparkSqlAstBuilder extends AstBuilder {
     ShowCurrentNamespaceCommand()
   }
 
+  /**
+   * Create a [[SetCatalogCommand]] logical command.
+   */
+  override def visitSetCatalog(ctx: SetCatalogContext): LogicalPlan = withOrigin(ctx) {
+    val name =
+      if (ctx.catalogIdentifier().identifier() != null) {
+        ctx.catalogIdentifier().identifier().getText
+      } else if (ctx.catalogIdentifier.STRING() != null) {
+        string(ctx.catalogIdentifier().STRING())
+      } else {
+        ctx.catalogIdentifier().getText
```

Comment:
We should never hit here. How about throw `IllegalStateException` here?

Reply:
OK, How about `QueryParsingErrors.invalidCatalogNameError(ctx)` with `ParseException` ?

Reply:
It's not a user-facing error, so we shouldn't use `QueryParsingErrors`